src/pages/user/settings/application.tsx

```diff
@@ -100,20 +101,35 @@ const AppSettingsPage = () => {
     setParams({ [UserDataKeys.EXTERNAL_LINK_ACTION]: id });
   };
 
-  const handleApplyDatabases = (names: string[]) => {
-    const newValue = JSON.parse(
-      JSON.stringify(settings.defaultDatabase),
-    ) as IADSApiUserDataResponse[UserDataKeys.DEFAULT_DATABASE];
-    newValue.forEach((v) => {
-      if (names.findIndex((n) => n === v.name) === -1) {
-        v.value = false;
-      } else {
-        v.value = true;
+  const handleApplyDatabases = useCallback(
+    (names: string[]) => {
+      const currentDatabases = settings[UserDataKeys.DEFAULT_DATABASE];
+
+      // if ALL is selected, reset to default (except for All)
+      if (names.includes(DatabaseEnum.All)) {
+        return setParams({
+          [UserDataKeys.DEFAULT_DATABASE]: [
+            ...DEFAULT_USER_DATA[UserDataKeys.DEFAULT_DATABASE],
```

Comment:
The default data already has `{ name: DatabaseEnum.All, value: false }`, so here the result is 
```[
  {
    "name": "All",
    "value": false
  },
  {
    "name": "Physics",
    "value": false
  },
  {
    "name": "Astronomy",
    "value": false
  },
  {
    "name": "General",
    "value": false
  },
  {
    "name": "Earth Science",
    "value": false
  },
  {
    "name": "All",
    "value": true
  }
]```

Reply:
Yep good catch, this should be fixed